### DISCUSSION TITLE
Can chezmoi sync dotfiles to a machine without internet connection/without a git repo?

### DISCUSSION THREAD

**Discussion by halloleo** [twpayne/chezmoi#2232] [Q&A] [answered]
New to chezmoi. I have looked through the User guide, but I'm still not sure whether I can use chezmoi _without_ a git repo.

**My scenario:** I use a Linux box without internet connection whose dotfiles I'd like to sync with my main internet connnected machines. I can move zip archives to the Linux box.

Is it still possible use chezmoi to syns the changed files and symlinks?

Thanks for any pointers!

**Comment by twpayne** [comment, marked as answer]
Yes, you can copy chezmoi and chezmoi's source state to your air-gapped box and run `chezmoi apply`. You'll be responsible for keeping the source state up-to-date, for example by manually transferring it from another machine. You won't be able to use externals, but normal files, directories, symlinks, and scripts should work.

Another option is to use [`chezmoi archive`](https://www.chezmoi.io/reference/commands/archive/) to generate the dotfiles on another machine and unpack that archive on your air-gapped machine.

**Reply by halloleo** [reply]
Thanks @twpayne. Will look in more detail at the `apply` command.
